### PR TITLE
Add default logging configuration for state machine

### PR DESCRIPTION
**Pull request by giograno**
Should address #6486. 
Passing an empty dictionary as `loggingConfiguration`, while being a valid AWS request, as shown in the ticker messes up with persistence. This PR adds the default configuration as the real AWS.

### CODE REVIEW

**Review by dominikschubert — CHANGES_REQUESTED** [review]
Needs an integration test but otherwise LGTM :+1:

**Review by whummer — APPROVED** [review]
As briefly discussed, mid-term we could consider registering the missing class in our upstream SFn persistence implementation. 👍  But the change LGTM, to unblock our users for now 🚀

File: tests/integration/test_stepfunctions.py
Comment (dominikschubert) [inline tests/integration/test_stepfunctions.py:611]: unsafe cleanup

This obv. isn't super critical when running against LocalStack but a test against AWS might leak resources here.

Either fixtures or other safer cleanup strategies should be used instead. A finally block would be ok too but would again need to wrap the individual cleanup statements in a safe way so they don't interfere with each other.

File: tests/integration/test_stepfunctions.py
Comment (dominikschubert) [inline tests/integration/test_stepfunctions.py:606]:
```suggestion
```

**Review by dominikschubert — APPROVED** [review]
minor nit regarding resource leaks in test, otherwise LGTM. Feel free to merge when this is fixed :+1:

**Pull request merged**
